### PR TITLE
Recreate #15175

### LINKED ISSUE #15172: Uncaught TypeError: Cannot read property 'createObjectStore' of undefined
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.18.0 x64
**Electron**: 1.3.15
**OS**: Microsoft Windows 7 Ultimate 
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'createObjectStore' of undefined

```
At C:\Users\Midhu\AppData\Local\atom\app-1.18.0\resources\app\src\state-store.js:17

TypeError: Cannot read property 'createObjectStore' of undefined
    at IDBOpenDBRequest.dbOpenRequest.onupgradeneeded (~/AppData/Local/atom/app-1.18.0/resources/app/src/state-store.js:17:19)
```

### Commands

```
```

### Non-Core Packages

```

```

### PR DESCRIPTION
**Pull request by icecream17**
See #15175
Resolves #15172 (although it's already closed anyways)

### CODE REVIEW

**Review by sadick254 — COMMENTED** [review]
The changes look good. Do we know what events that lead to the error?

**Comment by icecream17** [comment]
From what I gather about #15172 we don't know why the error happened.

**Comment by sadick254** [comment]
I think a good approach would be to display the error as a notification. This way we can have visibility when the error occurs. A user will be able to create an issue from the notification modal. See https://flight-manual.atom.io/api/v1.57.0/NotificationManager/

**Comment by icecream17** [comment]
Is the console.error unnecessary now that there's a notification?

**Comment by sadick254** [comment]
The `console.error` is fine.

**Comment by UziTech** [comment]
I think [`addFatalError`](https://flight-manual.atom.io/api/v1.57.0/NotificationManager/#instance-addFatalError) might be better so the button to easily create an issue appears.

File: src/state-store.js [hunk 1]
```diff
@@ -13,13 +13,24 @@ module.exports = class StateStore {
         const dbOpenRequest = indexedDB.open(this.databaseName, this.version);
         dbOpenRequest.onupgradeneeded = event => {
           let db = event.target.result;
+          db.onerror = error => {
+            atom.notifications.addError('Error loading database', {
```
Comment (sadick254) [inline src/state-store.js:17]:
```suggestion
            atom.notifications.addFatalError('Error loading database', {
```

File: src/state-store.js [hunk 2]
```diff
@@ -13,13 +13,24 @@ module.exports = class StateStore {
...
+            console.error('Error loading database', error);
+          };
           db.createObjectStore('states');
         };
         dbOpenRequest.onsuccess = () => {
           this.connected = true;
           resolve(dbOpenRequest.result);
         };
         dbOpenRequest.onerror = error => {
+          atom.notifications.addError('Could not connect to indexedDB', {
```
Comment (sadick254) [inline src/state-store.js:30]:
```suggestion
          atom.notifications.addFatalError('Could not connect to indexedDB', {
```

**Review by sadick254 — CHANGES_REQUESTED** [review]
I agree with @UziTech

**Review by sadick254 — APPROVED** [review]

**Pull request merged**